### PR TITLE
Feat/commit task

### PR DESCRIPTION
**Pull request by antmat**
This PR adds ability to optionally commit container as task_id tag back to registry
@noxiouz @3Hren PTAL

### CODE REVIEW

File: insonmnia/miner/container.go [hunk 1]
```diff
@@ -6,12 +6,14 @@ import (
...
 	"go.uber.org/zap"
 
+	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/network"
 	"github.com/docker/docker/client"
 	"github.com/gliderlabs/ssh"
 	log "github.com/noxiouz/zapctx/ctxlog"
+	"io"
```
Comment (3Hren) [inline insonmnia/miner/container.go]: Put on top of the import.

File: insonmnia/miner/container.go [hunk 2]
```diff
@@ -172,3 +170,58 @@ func containerRemove(ctx context.Context, client client.APIClient, id string) {
 		log.G(ctx).Error("failed to remove the container", zap.String("id", id), zap.Error(err))
 	}
 }
+
+func (c *containerDescriptor) upload() error {
+	opts := types.ContainerCommitOptions{}
+	resp, err := c.client.ContainerCommit(c.ctx, c.ID, opts)
+	log.G(c.ctx).Info("commit")
```
Comment (3Hren) [inline insonmnia/miner/container.go]: Commit what?

File: insonmnia/miner/overseer.go [hunk 3]
```diff
@@ -224,16 +226,26 @@ func (o *overseer) handleStreamingEvents(ctx context.Context, sinceUnix int64, f
...
-				if cok {
-					c.remove()
-				} else {
+				if !cok {
 					// NOTE: it could be orphaned container from our previous launch
 					log.G(ctx).Warn("unknown container with sonm tag will be removed", zap.String("id", id))
 					containerRemove(o.ctx, o.client, id)
+					continue
+				}
+				if sok {
```
Comment (3Hren) [inline insonmnia/miner/overseer.go]: U sok cok?

File: insonmnia/miner/container.go [hunk 4]
```diff
@@ -172,3 +170,58 @@ func containerRemove(ctx context.Context, client client.APIClient, id string) {
...
+		return err
+	}
+
+	newImg, err := reference.WithTag(named, c.description.TaskId)
+	if err != nil {
+		log.G(c.ctx).Error("failed to add tag", zap.String("id", resp.ID), zap.Error(err))
+		return err
+	}
+
+	log.G(c.ctx).Info("tagging image", zap.String("from", resp.ID), zap.Any("to", newImg))
```
Comment (noxiouz) [inline insonmnia/miner/container.go]: zap.Stringer vs Any

**Pull request merged**